output/transform/util.go

```diff
@@ -82,10 +91,15 @@ func upsertQueryReferenceAndInformationSimple(server *state.Server, refs []*snap
 	idx := int32(len(refs))
 	refs = append(refs, &newRef)
 
+	normalizedQuery := util.NormalizeQuery(originalQuery, server.Config.FilterQueryText, trackActivityQuerySize)
+	if normalizedQuery == util.QueryTextTruncated {
+		normalizedQuery = ""
+	}
```

Comment:
I don't follow this change to the truncated query handling. I see you replace the marker text with the empty string, but I don't understand what impact that has. Should we add a comment here? Or should this be happening in NormalizeQuery, and we can explain it there?

Reply:
>I see you replace the marker text with the empty string, but I don't understand what impact that has.

The internal PR this is paired with skips over queries whose query text is empty. It could just as well be a new query text placeholder to self-document the behavior, I just didn't want to send extra bytes that are going to be discarded.

>Or should this be happening in NormalizeQuery, and we can explain it there?

I kept it here to avoid changing the full snapshot behavior, which also uses `util.NormalizeQuery`

Reply:
Got it. I think it's okay to use the empty string as a sentinel, but maybe we should add a comment here? It's pretty hard to tell why this is flagged as empty text.

Reply:
Reviewing this again, I'm not really sure why I made this change. Since we're discussing elsewhere that we may want to stop storing queries using placeholder text, it may be better to keep this as `<truncated query>`